plugins/kubernetes/app/models/kubernetes/release.rb

```diff
@@ -24,12 +24,11 @@ def user
     # Creates a new Kubernetes Release and corresponding ReleaseDocs
     def self.build_release_with_docs(params)
       roles = params.delete(:grouped_deploy_group_roles).to_a
-      release = new(params) do |release|
-        if roles.flatten(1).any? { |dgr| dgr.kubernetes_role.blue_green? }
-          release.blue_green_color = begin
-            release.previous_succeeded_release&.blue_green_color == "blue" ? "green" : "blue"
-          end
-        end
+      release = new(params) do |r|
+        # We always set the blue green color but the template filler might not always use it.
+        # This allows us to support deploy-specific features like env vars from config maps.
+        r.blue_green_color =
+          r.previous_succeeded_release&.blue_green_color == "blue" ? "green" : "blue"
```

Comment:
```suggestion
      release = new(params) do |release|
        # We always set the blue green color but the template filler might not always use it.
        # This allows us to support deploy-specific features like env vars from config maps.
        release.blue_green_color =
          release.previous_succeeded_release&.blue_green_color == "blue" ? "green" : "blue"
```

Reply:
I don't really like shadowing variables here, which is why I left it as a single letter, but I'll work something out.

Reply:
ah yeah, did not see that ... `r` is fine then, shadowing in this way seems alright to me too, so do whatever